Admin/AdminDefintionInterface.php

```diff
@@ -0,0 +1,123 @@
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Knp\Menu\ItemInterface as MenuItemInterface;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Sonata\AdminBundle\Show\ShowMapper;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+interface AdminDefintionInterface
+{
+    /**
+     * Returns the array of allowed export formats.
+     *
+     * @return array
+     */
+    public function getExportFormats();
+
+    /**
+     * @param mixed $object
+     */
+    public function preValidate($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function preUpdate($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function postUpdate($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function prePersist($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function postPersist($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function preRemove($object);
+
+    /**
+     * @param mixed $object
+     */
+    public function postRemove($object);
+
+    /**
+     * Call before the batch action, allow you to alter the query and the idx.
+     *
+     * @param string              $actionName
+     * @param ProxyQueryInterface $query
+     * @param array               $idx
+     * @param bool                $allElements
+     */
+    public function preBatchAction($actionName, ProxyQueryInterface $query, array &$idx, $allElements);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureFormFields(FormMapper $form);
+
+    /**
+     * @param ListMapper $list
+     */
+    public function configureListFields(ListMapper $list);
+
+    /**
+     * @param DatagridMapper $filter
+     */
+    public function configureDatagridFilters(DatagridMapper $filter);
+
+    /**
+     * @param ShowMapper $show
+     */
+    public function configureShowFields(ShowMapper $show);
+
+    /**
+     * @param RouteCollection $collection
+     */
+    public function configureRoutes(RouteCollection $collection);
+
+    /**
+     * Allows you to customize batch actions.
+     *
+     * @param array $actions List of actions
+     *
+     * @return array
+     */
+    public function configureBatchActions($actions);
+
+    /**
+     * Configures the tab menu in your admin.
+     *
+     * @param MenuItemInterface $menu
+     * @param string            $action
+     * @param AdminInterface    $childAdmin
+     *
+     * @return mixed
+     */
+    public function configureTabMenu(MenuItemInterface $menu, $action, AdminInterface $childAdmin = null);
```

Comment:
I mean that there are some methods that are overriden all the time, in every admin, and other that are not. This one is only needed when you have custom actions on a given admin, so yeah not  rare, but not overriden all the time either (so, kind of optional)